static/src/javascripts/projects/common/modules/experiments/tests/membership-engagement-immediate.js

```diff
@@ -0,0 +1,114 @@
+define([
+    'bean',
+    'qwery',
+    'common/utils/config',
+    'common/utils/storage',
+    'common/utils/template',
+    'common/modules/ui/message',
+    'text!common/views/membership-message.html',
+    'common/modules/commercial/commercial-features',
+    'common/utils/mediator'
+], function (
+    bean,
+    qwery,
+    config,
+    storage,
+    template,
+    Message,
+    messageTemplate,
+    commercialFeatures,
+    mediator
+) {
+    return function () {
+
+        this.id = 'MembershipEngagementImmediate';
+        this.start = '2016-10-13';
+        this.expiry = '2016-10-15';
+        this.author = 'Justin Pinner';
+        this.description = 'Test if showing engagement banner to users with less than 10 page views drives additional contributions.';
+        this.audience = 0.2;
+        this.audienceOffset = 0;
+        this.successMeasure = 'Conversion for membership';
+        this.showForSensitive = false;
+        this.audienceCriteria = 'Twenty percent of uk edition readers (desktop and mobile web)';
+        this.dataLinkNames = '';
+        this.idealOutcome = '';
+        this.hypothesis = 'Although the conversion rate will likely drop slightly, we may see an uplift in the absolute value of contributions due to reaching a larger, and perhaps new audience segment.';
+
+        this.canRun = function () {
+            var matchesEdition = config.page.edition.toLowerCase() === 'uk';
+            return commercialFeatures.canReasonablyAskForMoney && matchesEdition;
+        };
+
+        var makeMessage = function(id) {
+            var linkHref = 'https://membership.theguardian.com/uk/supporter?INTCMP=' + id;
+            var colours = ['default','vibrant-blue','yellow','light-blue','deep-purple','teal'];
+            // Rotate through different colours on successive page views
+            var colourIndex = storage.local.get('gu.alreadyVisited') % colours.length;
+            var cssModifierClass = 'membership-message-' + colours[colourIndex];
+            var messageText = 'For less than the price of a coffee a week, you could help secure the Guardian’s future. Support our journalism for just £49 per year.';
+            var renderedBanner = template(messageTemplate, {messageText: messageText, linkHref: linkHref});
+            return new Message(
+                // change this to redisplay banners to everyone who has previously closed them
+                // 2016-10-12 matches the current PROD message id
+                'engagement-banner-2016-10-12',
+                {
+                    pinOnHide: false,
+                    siteMessageLinkName: 'membership message',
+                    siteMessageCloseBtn: 'hide',
+                    siteMessageComponentName: id,
+                    trackDisplay: true,
+                    cssModifierClass: cssModifierClass
+                }).show(renderedBanner);
+
+        };
+
+        this.variants = [
```

Comment:
This is a lot of changes to make in this file, when it might be easier (and less code) to use [this](https://github.com/guardian/frontend/pull/10916/files) as an example of how to change your code by variant (if statements) instead of doing it in this file